client/src/GameState.ts

```diff
@@ -19,7 +19,11 @@ export class GameState {
     // i.e. if you are player 1, these are of player 2, then 3, then 0
     otherTetrominoes: Array<Tetromino>;
     playerId!: 0 | 1 | 2 | 3;
+    randomBag: RandomBag;
 
+    public getNewPiece(): TetrominoType {
```

Comment:
Why do we need a separate function for getting the next piece when the RandomBag function already handles this? Couldn't SceneGameArena just call `this.gameState.randomBag.returnNextPiece();`?

Idk if that's any better though lol. Whats the input from others?

Reply:
Yeah it can just call the random bag class, but I just put that there as an abstraction.